builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1488,9 +1505,34 @@ func (vm *virtualMachine) setupVirtualMachine(c *govmomi.Client) error {
 	}
 
 	var resourcePool *object.ResourcePool
-	if vm.resourcePool == "" {
-		if vm.cluster == "" {
-			resourcePool, err = finder.DefaultResourcePool(context.TODO())
+	var hostSystem *object.HostSystem
+	if vm.hostSystem == "" {
+		if vm.resourcePool == "" {
+			if vm.cluster == "" {
+				resourcePool, err = finder.DefaultResourcePool(context.TODO())
+				if err != nil {
+					return err
+				}
+				hostSystem, err = finder.DefaultHostSystem(context.TODO())
+				if err != nil {
+					return err
+				}
+			} else {
+				resourcePool, err = finder.ResourcePool(context.TODO(), "*"+vm.cluster+"/Resources")
```

Comment:
Are we using `vm.cluster+"/Resources` on other `finders`, or is this specific to this use case?